### PR TITLE
refactor: migrate tests to testing-library

### PR DESCRIPTION
**Pull request by PatrykBuniX**
- migrate last portion of tests to testing library,
- fix some of the ts strict errors along the way
- delete TestPage util 😄

### CODE REVIEW

File: src/script/components/AvailabilityState.test.tsx [hunk 1]
```diff
@@ -18,67 +18,80 @@
...
+  label: 'example',
+  showArrow: false,
+  theme: false,
+};
 
 describe('AvailabilityState', () => {
   it('renders available icon', async () => {
-    const availabilityState = new AvailabilityStatePage({
+    const props = {
+      ...defaultProps,
```
Comment (przemvs) [inline src/script/components/AvailabilityState.test.tsx]: Hmm.. I think you don't need pass there availability, you have it in defaultProps :)

**Review by przemvs — APPROVED** [review]
LGTM! :) 1 small thing only :)

File: src/script/components/AvailabilityState.test.tsx (hunk 1 above)
Comment (PatrykBuniX) [reply src/script/components/AvailabilityState.test.tsx]: done 👍

**Review by atomrc — APPROVED** [review]
💪

**Pull request merged**